### PR TITLE
Restructure `semigroupoid` package into `simple-semigroupoids` package.

### PR DESCRIPTION
**Pull request by jorisdral**
The `semigroupoid` package has been renamed to `simple-semigroupoids`, which should make clear that it is different from the `semigroupoids` package that is currently on Hackage. Furthermore, we have renamed/restructured the modules to prevent module name clashes with the `semigroupoids` package.

### CODE REVIEW

File: simple-semigroupoids/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# simple-semigroupoids
```
Comment (Jasagredo) [inline simple-semigroupoids/README.md:1]: Perhaps worth adding some information here?

File: simple-semigroupoids/simple-semigroupoids.cabal [hunk 2]
```diff
@@ -15,12 +15,12 @@ build-type:          Simple
 
 
 library
-  default-language:    Haskell2010
+  default-language:    GHC2021
```
Comment (Jasagredo) [inline simple-semigroupoids/simple-semigroupoids.cabal:18]: is this already supported in 8.10.7?

**Review by Jasagredo — APPROVED** [review]
Looks good!

File: simple-semigroupoids/simple-semigroupoids.cabal (hunk 2 above)
Comment (jorisdral) [reply simple-semigroupoids/simple-semigroupoids.cabal]: You're right, that's an oversight. I've reverted this change

File: simple-semigroupoids/README.md (hunk 1 above)
Comment (jorisdral) [reply simple-semigroupoids/README.md:1]: I plan on doing that at a later stage, but definitely before releasing the package on CHaP!

**Pull request merged**